misc/sudokusolver.c

```diff
@@ -0,0 +1,77 @@
+//recursion problem : Sudoku Solver
+/*You are given an incomplete N*N Sudoku and asked to solve it using the following recursive algorithm:
+(1) Scan the Sudoku from left to right row-wise to search for an empty cell.
+(2) If there are no empty cells, print the Sudoku. Go to step 5.
+(3) In the empty cell, try putting numbers 1 to N while ensuring that no two numbers in a single row, column, or box are same. Go back to step 1.
+(4) Declare that the Sudoku is Invalid.
+(5) Exit.*/
+
+#include <stdio.h>
+
+const int M=144;
+int N, R, C;
+
+int OKrow(int a[M], int x, int y, int v) {
+	int j;
+	for(j=0; j<N; j++)
+		if(a[x*N+j] == v)
+			return 0;
+	return 1;
+}
+int OKcol(int a[M], int x, int y, int v) {
+	int i;
+	for(i=0; i<N; i++)
+		if(a[i*N+y] == v)
+			return 0;
+	return 1;
+}
+int OKbox(int a[M], int x, int y, int v) {
+	int bi=x/R, bj=y/C, i, j;
+	for(i=0; i<R; i++)
+		for(j=0; j<C; j++)
+			if(a[(i + bi*R)*N + (j + bj*C)] == v)
```

Comment:
isnt this counter intuitive to line# 29?
`bi*R` operation on this line undoes the operation on line #29 and is always equal to `x`!